src/v/features/feature_table.h

```diff
@@ -422,6 +422,12 @@ inline constexpr std::array feature_schema{
     feature::datalake_iceberg_ga,
     feature_spec::available_policy::always,
     feature_spec::prepare_policy::always},
+  feature_spec{
+    release_version::v25_1_1,
+    "cloud_storage_read_write_fence",
+    feature::cloud_storage_metadata_rw_fence,
```

Comment:
@Lazin how did you discover this? it feels like this implies that there is a hole in upgrade testing for the fencing feature?

Reply:
Spotted this during code review. This shows that we don't have a single unit tests proving that fencing works?

Reply:
> how did you discover this?
It's not me, it's @oleiman 

> This shows that we don't have a single unit tests (integration/fixture test with feature table?) proving that fencing works?
We have upgrade tests which are running mixed version Redpanda.
These tests were failing without the feature flag, they stopped failing when I added the flag without adding a `feature_spec`. It looks like the spec is used for admin api and maybe for something else but it's not critical  for correctness.

Reply:
> not critical for correctness.

Without this I reckon the feature was never enabled. Right? So how it is not critical for correctness if we haven't emitted fences prior to this PR? Or I have it wrong?

Reply:
The feature table is definitely not working correctly (I think snapshotting depends on this) but I was able to trigger a known bug in mixed version test. The bug can only be triggered by rw-fence command.

Reply:
@Lazin @nvartolomei 

>The feature table is definitely not working correctly

what does this mean? is this an unresolved bug we should be concerned about?